data_handler/handlers/order_books/myswap/main.py

```diff
@@ -36,18 +37,16 @@ def __init__(self, base_token: str, quote_token: str, apply_filtering: bool = Fa
         self.connector = MySwapAPIConnector()
         self.apply_filtering = apply_filtering
         self.logger = get_logger("MySwap", Path.cwd().joinpath("./logs"))
-        self.base_token_name = self.get_token_configs()[0].name
         self._decimals_diff = Decimal(10 ** (self.token_a_decimal - self.token_b_decimal))
+        self._normalize_addresses()
 
-    def _read_liquidity_data(self, pool_id: str) -> pd.DataFrame:
-        """
-        Read liquidity data from the MySwap data service.
-        :param pool_id: str - The pool id in hexadecimal.
-        :return: pd.DataFrame - The liquidity data.
-        The structure of the data:
-        Columns: tick(numpy.int64), liq(numpy.int64)
-        """
-        return pd.read_json(self.MYSWAP_URL.format(pool_id=pool_id), compression="gzip")
+    def _normalize_addresses(self) -> None:
+        """Remove leading zeroes from token addresses. Raise Value Error if address can't be converted to int."""
```

Comment:
Why do we have to remove the leading zeroes? Imo addresses **with** leading zeroes are considered normalized 😉

Reply:
We need to remove them because AMM process addresses without leading zeroes. Changed method name, and now it returns new addresses instead of replacing old ones.

Reply:
Ok, and do you know if there's a reason for the AMM processing to work with addresses without leading zeros. Ideally, I'd always like to work will complete addresses (with leading zeroes).

No need to make any adjustments now, I'm just curious if there's a specific reason for that.

Reply:
I don't know the reason exactly, but I'm guessing it's because addresses are stringified as is, so without leading zeroes added. It would make more work if we needed to convert 0x123 to 0x0123 before response from backend